value/listreflect.go

```diff
@@ -40,3 +40,33 @@ func (r listReflect) Unstructured() interface{} {
 	}
 	return result
 }
+
+func (r listReflect) Range() ListRange {
+	return &listReflectRange{r.Value, newTempValuePooler(), -1, r.Value.Len()}
```

Comment:
Note that `listReflectRange` will escape here too

Reply:
Sure, I've pooled both together. It works out nicely, actually.